src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutValenceCheck.java

```diff
@@ -0,0 +1,167 @@
+ * Wiki, each roundabout should have greater than 1 connection as 1 connection should be tagged as a
+ * turning point, and no connections is obviously not a valid way.
+ *
+ * @author savannahostrowski
+ */
+public class RoundaboutValenceCheck extends BaseCheck
+{
+
+    private static final long serialVersionUID = 1L;
+    public static final String WRONG_VALENCE_INSTRUCTIONS = "This roundabout, {0,number,#}, "
+            + "has the wrong valence. It has a valence of {1,number,#}.";
+    public static final String VALENCE_OF_ONE_INSTRUCTIONS = "This feature, {0,number,#},"
+            + " should not be labelled as a roundabout. "
+            + "This feature should be a turning loop or turning circle.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList(WRONG_VALENCE_INSTRUCTIONS, VALENCE_OF_ONE_INSTRUCTIONS);
+
+    private static final double LOWER_VALENCE_THRESHOLD_DEFAULT = 2.0;
+    private static final double UPPER_VALENCE_THRESHOLD_DEFAULT = 10.0;
+    private final double minimumValence;
+    private final double maximumValence;
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public RoundaboutValenceCheck(final Configuration configuration)
+    {
+        super(configuration);
+
+        this.minimumValence = (double) configurationValue(configuration, "connections.minimum",
+                LOWER_VALENCE_THRESHOLD_DEFAULT);
+        this.maximumValence = (double) configurationValue(configuration, "connections.maximum",
+                UPPER_VALENCE_THRESHOLD_DEFAULT);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // We check that the object is an instance of Edge
+        return object instanceof Edge
+                // And that the Edge has not already been marked as flagged
+                && !this.isFlagged(object.getIdentifier())
+                // Make sure that the edges are instances of roundabout
+                && JunctionTag.isRoundabout(object);
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+
+        // Get all edges in the roundabout
+        final Set<Edge> roundaboutEdges = getAllRoundaboutEdges(edge);
+
+        final Set<Edge> connectedEdges = roundaboutEdges.stream()
+                .flatMap(roundaboutEdge -> roundaboutEdge.connectedEdges().stream())
+                .filter(HighwayTag::isCarNavigableHighway).filter(Edge::isMasterEdge)
+                .filter(e -> !JunctionTag.isRoundabout(e)).filter(e -> !roundaboutEdges.contains(e))
```

Comment:
Please avoid using single letter variables to keep styling consistent within code base.